project/backend/src/test/java/com/google/sps/TestRestaurantParser.java

```diff
@@ -0,0 +1,43 @@
+package com.google.sps;
+
+import com.google.sps.data.JsonToRestaurantParser;
+import com.google.sps.data.Restaurant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class TestRestaurantParser {
+
+  private static final int MISSING_FIELD = -1;
+  private static final List<String> EMPTY_TYPES = new ArrayList<>();
+  private static final String RESTAURANT_A = "RESTAURANT_A";
+  private static final String PLACE_ID_A = "A12345";
+  private static final String VICINITY_A = "10 Main Street";
+  private static final Map<String, Double> COORDS_A = Map.ofEntries(
+    Map.entry("lat", 10.0),
+    Map.entry("lng", 20.0)
+  );
+
+  @Test
+  public void noOptionalFields() throws JSONException {
```

Comment:
Also I believe this exception should be caught

Reply:
nah, so for tests, you usually just let this be and the caller will handle the exception, a test that should not throw an exception, will just pass if what you're trying to assert at the end passes

Reply:
you should have a separate test where you assert that the exception is thrown to test that case (Where you give it a malformed input)